source/mqtt/MqttClient.cpp

```diff
@@ -674,9 +682,12 @@ namespace Aws
                 opCompleteCallbackData->allocator = m_owningClient->allocator;
                 opCompleteCallbackData->onOperationComplete = std::move(onOpComplete);
                 opCompleteCallbackData->topic = topicCpy;
+                opCompleteCallbackData->payload = Crt::New<ByteBuf>(m_owningClient->allocator);
```

Comment:
redundant: assuming payload isn't a `ByteBuf *` then this goes away entirely. So does the call to check if it is null before calling cleanup. Just call cleanup.